### DISCUSSION TITLE
flag-based purge presets in `tailwindcss init`

### DISCUSSION THREAD

**Discussion by smhmd** [tailwindlabs/tailwindcss#2627] [Ideas]
Add `purge` array based on flags passed to `tailwindcss init`. For instance, invoking:

```fish
$ npx tailwindcss init -p --vue # or --vue-cli
```
or
```fish
$ npx tailwindcss init -p --purge=vue
```
Could result in `tailwind.config.js` having a `purge` property's value of:
```js
purge: [
  './public/**/*.html',
  './src/**/*.html',
  './src/**/*.vue',
],
```
There could be other presets for `vite`, `nuxt`, `react`, `next`, etc.

**Comment by simonswiss** [comment]
We had an internal discussion this week about having a `purge by default` setup, which would involve some sensible default/presets like you mention.

I think this is definitely an interesting approach. It's not a lot of work to add the paths manually, but removing any friction point towards a "purged out of the box" Taliwind sounds like a good plan to me 👍
